archive/j/java/Factorial.java

```diff
@@ -0,0 +1,29 @@
+public class Factorial
+{
+    public static long fact(long n)
+    {
+        if (n <= 0)
+            return 1;
+        return n * fact(n - 1);
+    }
+
+    public static void main(String[] args)
+    {
+        try
+        {
+            long n = Long.parseLong(args[0]);
+            if (n > 59)
+            {
+                System.out.println(String.format("%1$s! is out of the reasonable bounds for calculation.", n));
+                System.exit(1);
+            }
+            long result = fact(n);
+            System.out.println(result);
+        }
+        catch (Exception e)
```

Comment:
I thought there were a couple to catch here, but since we have the check for the size of the input, I suppose the only issue here would be with the parsing. I changed it to NumberFormatException.